packages/nodes-base/nodes/Grist/Grist.node.ts

```diff
@@ -0,0 +1,247 @@
+
+					// https://support.getgrist.com/api/#tag/records/paths/~1docs~1{docId}~1tables~1{tableId}~1records/patch
+
+					const body = { records: [ { fields: {} } ] } as GristUpdateRowPayload;
+
+					const rowId = this.getNodeParameter('rowId', i) as string;
+					const dataToSend = this.getNodeParameter('dataToSend', 0) as 'defineBelow' | 'autoMapInputData';
+
+					if (dataToSend === 'autoMapInputData') {
+						const incomingKeys = Object.keys(items[i].json);
+						const rawInputsToIgnore = this.getNodeParameter('inputsToIgnore', i) as string;
+						const inputDataToIgnore = rawInputsToIgnore.split(',').map(c => c.trim());
+
+						const acc: IDataObject = {};
+
+						for (const key of incomingKeys) {
+							if (inputDataToIgnore.includes(key)) continue;
+							acc[key] = items[i].json[key];
+						}
+
+						body.records[0].fields = acc;
+
+					} else if (dataToSend === 'defineBelow') {
+						const { properties } = this.getNodeParameter('fieldsToSend', i, []) as {
+							properties: GristDefinedData;
+						};
+
+						if (!properties.length) {
+							throw new NodeOperationError(this.getNode(), 'Please enter data to send');
+						}
+
+						body.records = [
+							{ fields: parseFieldsToSend(properties) },
+						];
+					}
+
+					body.records = body.records.map(record => {
+						return {
+							id: Number(rowId),
+							...record,
+						};
+					});
+
+					const docId = this.getNodeParameter('docId', i) as string;
+					const tableId = this.getNodeParameter('tableId', i) as string;
+					const endpoint = `/docs/${docId}/tables/${tableId}/records`;
+
+					await gristApiRequest.call(this, 'PATCH', endpoint, body);
+					responseData = { success: true };
+
+				} else if (operation === 'getAll') {
+
+					// ----------------------------------
+					//             getAll
+					// ----------------------------------
+
+					// https://support.getgrist.com/api/#tag/records
+
+					const docId = this.getNodeParameter('docId', i) as string;
+					const tableId = this.getNodeParameter('tableId', i) as string;
+					const endpoint = `/docs/${docId}/tables/${tableId}/records`;
+
+					const qs: IDataObject = {};
+
+					const returnAll = this.getNodeParameter('returnAll', i) as boolean;
+
+					if (!returnAll) {
+						qs.limit = this.getNodeParameter('limit', i) as number;
+					}
+
+					const { sort, filter } = this.getNodeParameter('additionalOptions', i) as GristGetAllOptions;
+
+					if (sort?.sortProperties.length) {
+						qs.sort = parseSortProperties(sort.sortProperties);
+					}
+
+					if (filter?.filterProperties.length) {
+						const parsed = parseFilterProperties(filter.filterProperties);
+						qs.filter = JSON.stringify(parsed);
+					}
+
+					responseData = await gristApiRequest.call(this, 'GET', endpoint, {}, qs);
+					responseData = responseData.records;
+				}
+
+			} catch (error) {
+				if (this.continueOnFail()) {
+					returnData.push({ json: { error: error.message } });
```

Comment:
It's not clear to me when `json` should be used if already using `returnJsonArray`? I tried a request which failed on one item and this is what the JSON output looks like:

```json
[
  {
    "id": 2
  },
  {
    "json": {
      "error": "The service was not able to process your request"
    }
  }
]
```